juju/juju
# Add new update-k8s command

## Description of change

Add a new update-k8s command.
The command can either accept a built-in cloud name like "microk8s", or else
behaves the same as update-cloud where a cloud file is used to pass in the updated info.

The CloudFileReader from cmd/juju/cloud/add.go was reused so was exported. The CloudMetadataStore used in the cmd/juju/cloud had a ParseCloudMetadataFile() method but this caused duplicate test data to be needed in tests/mocks and didn't work well with the new cass/update-k8s tests. So it was reworked to have a ReadCloudData() method instead.

Also remove from unused code from update-cloud.
And a driveby fix for https://bugs.launchpad.net/juju/+bug/1865894

## QA steps

Run the command for microk8s and also a bespoke clouds.yaml file
$ juju update-k8s microk8s

$ juju update-k8s mytestk8s -f clouds.yaml

Do it for both --client and a k8s added to a lxd controller
Check cloud is updated with show-cloud

## Bug reference

https://bugs.launchpad.net/juju/+bug/1864259
